### ISSUE TITLE
setPreventInvalidInput(true) on EmailField prevents the user to type in anything

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow-components#2986]
### Description

setPreventInvalidInput(true) on EmailField prevents the user to type in anything. E.g. Pressing `a` has no effect (well the cursor goes red briefly) since it's not a valid e-mail address.

### Expected outcome

I should be able to gradually type in e-mail address.

### Actual outcome

I can't type in anything.

### Minimal reproducible example

```java
@Route("")
public class MainView extends VerticalLayout {

    public MainView() {
        final EmailField emailField = new EmailField("E-mail");
        emailField.setPreventInvalidInput(true);
        add(emailField);
    }
}
```

### Steps to reproduce

1. Paste the above to Skeleton starter and run it
2. Start typing text into EmailField - nothing happens.

### Environment

- Vaadin 14.8.7
- Ubuntu 21.10

### Browsers Affected
- [ ] Chrome
- [x] Firefox 99
- [ ] Safari
- [ ] Edge
- [ ] IE 11
- [ ] iOS Safari
- [ ] Android Chrome

**Comment by TatuLund** [comment]
Yes, but this luckily seems to work. 

```
@Route("email")
public class EmailView extends VerticalLayout {
    
    public EmailView() {
        final EmailField emailField = new EmailField("E-mail");
        emailField.addValueChangeListener(e -> {
            if (!emailField.isInvalid()) {
                Notification.show(e.getValue());
            }
        });
        add(emailField);
    }
}
```

**Comment by TatuLund** [comment]
The web component seems to do the checking after every key stroke. But I think it should be done only when intended value change is to happen.

**Comment by sissbruecker** [comment]
After some discussion we decided that the API shouldn't be there in the first place, and that it should be removed. Instead developers should rely on the regular validation, or set custom pattern validation, and let users correct their changes.
Meanwhile we can deprecate the existing method, and add a note to the JavaDoc that it has no practical use for email field.

**Comment by mvysny** [comment]
I agree, but then what's the point of having an EmailField - what does it offer on top of a regular TextField?

**Comment by rolfsmeds** [comment]
With the introduction of `allowed-chars-pattern` (https://github.com/vaadin/platform/issues/2963), we could, instead of removing `setPreventInvalidInput`, re-define it to toggle the usage of an allowed-chars pattern pre-populated with chars valid in an email address. That way the Email Field would prevent users from entering characters that are invalid in emails when `setPreventInvalidInput` is true.

**Comment by web-padawan** [comment]
We removed `setPreventInvalidInput` API in V24 since it was essentially broken. Closing this as no longer relevant.
